### PR TITLE
Run clean feature added & fixed noDependencies argument typo

### PR DESCRIPTION
**Pull request by lucax88x**
added:
- run command
- clean command
- adjusted docs
- corrected noDepenencies to noDependencies typo

**User Story** (edited by [TerribleDev](https://github.com/terribledev))

As a user I'd like to use the builtin `dotnet clean` instead of importing del

* add a clean command which will call `dotnet clean [Project] [Options]`

example

```js
gulp.task('clean', ()=>{
    return gulp.src('**/*.csproj', {read: false})
               .pipe(clean({echo: true}));
})

gulp.task('build', ['clean'], ()=>{...});

```

OR:

```js
gulp.task('build', ()=>{
    return gulp.src('**/*.csproj', {read: false})
               .pipe(clean())
               .pipe(build());
})
```
As a user I'd like to use the builtin `dotnet run` command from gulp
* add a run command which will call `dotnet run [Project] [Options]`

example

```js
gulp.task('exec', ()=>{
    return gulp.src('**/*.csproj', {read: false})
               .pipe(run());
})

```

### CODE REVIEW

**Comment by TerribleDev** [comment]
Hi, how is this different than #12

**Comment by TerribleDev** [comment]
Thanks for the pr

**Comment by TerribleDev** [comment]
Hi @lucax88x the clean command is implemented in #12 What I would ask of you is to remove clean from your PR, and keep run. Also look to #12 for any thing we may have missed that you potentially wanted.

After you adjust your PR, I'll ensure both things get merged. I'll look it all over this weekend. I'm currently traveling, and won't be back in the office until Friday.

Thank you so much for your PR

**Comment by lucax88x** [comment]
No worries, thanks you for the awesome gulp extension! :)
I really missed the #12, I'll try to adjust the PR during the day removing the clean part then.

**Comment by lucax88x** [comment]
just saw your #12.

Isn't my PR the same but that adds more than yours? (like tests, documentations, etc.)

What about closing yours in favor of mine?

**Comment by TerribleDev** [comment]
That's fine by me....

File: lib/builders/restoreArgBuilder.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ module.exports = (value) => {
     if (value.noIncremental) {
         args = args.concat(['--no-incremental']);
     }
-    if (value.noDepenencies) {
```
Comment (TerribleDev) [inline lib/builders/restoreArgBuilder.js:28]: is this a bug fix?

File: test/gulpfile.js [hunk 2]
```diff
@@ -1,35 +1,41 @@
 let gulp = require('gulp');
-let {restore, build, test, pack, publish} = require('gulp-dotnet-cli');
+let { clean, restore, build, test, pack, publish, run } = require('../lib/index');
```
Comment (TerribleDev) [inline test/gulpfile.js:2]: good idea

**Comment by TerribleDev** [comment]
I like your PR. I just had a question or two. When I come back (Friday-ish) I'll merge and deploy this to npm.

File: lib/builders/restoreArgBuilder.js (hunk 1 above)
Comment (TerribleDev) [reply lib/builders/restoreArgBuilder.js:28]: if it is can you adjust the initial comment in this PR and add this bug to the list of things this PR comes with?

Comment (lucax88x) [reply lib/builders/restoreArgBuilder.js:28]: Yeah, you are right, it was a typo but also a bug

Comment (lucax88x) [reply lib/builders/restoreArgBuilder.js:28]: done!

**Pull request merged**